hnn_core/tests/test_parallel_backends.py

```diff
@@ -32,6 +33,43 @@ def _terminate_mpibackend(event, backend):
         sleep(0.01)
 
 
+def test_gid_assignment():
+    """Test that gids are assigned without overlap across ranks"""
+
+    net = jones_2009_model(add_drives_from_params=False)
+    weights_ampa = {'L2_basket': 1.0, 'L2_pyramidal': 3.0,
+                    'L5_basket': 2.0, 'L5_pyramidal': 4.0}
+    syn_delays = {'L2_basket': 1.0, 'L2_pyramidal': 2.0,
+                  'L5_basket': 3.0, 'L5_pyramidal': 4.0}
+
+    net.add_bursty_drive(
+        'bursty_dist', location='distal', burst_rate=10,
+        weights_ampa=weights_ampa, synaptic_delays=syn_delays,
+        cell_specific=False, n_drive_cells=5)
+    net.add_evoked_drive(
+        'evoked_prox', mu=1.0, sigma=1.0, numspikes=1,
+        weights_ampa=weights_ampa, location='proximal',
+        synaptic_delays=syn_delays, cell_specific=True,
+        n_drive_cells='n_cells')
+    net._instantiate_drives(tstop=20, n_trials=2)
+
+    n_cells = {type: len(gid_range) for type, gid_range in
```

Comment:
Can you test explicitly that there is indeed no overlap? I'm thinking of just aggregating a list of gids from all ranks, and ensuring that there are no repeat gids with `assert len(all_gids) == len(set(all_gids))`

Reply:
Should that be done here or in `test_network.py`? It probably doesn't matter a ton, but ensuring that the gids are created correctly in `net.gid_ranges` seems like a task for `Network` and `NetworkBuilder`.

Reply:
I understood that @ntolley was talking about `gid_list` not `gid_ranges`? The comment seems misplaced though ...

Reply:
Sorry the placing is strange, this just seems the logical place to collect the gids from all ranks. I was thinking you could set a variable here named something like `all_gids = list()` and in the loop over the ranks below you can 
```py
all_gids.extend(net_builder._gid_list)
```
Since the PR is fixing a problem where gids are overlapping across nodes, this directly checks if there are duplicate gids no?

Reply:
This a great point actually. WDYT of https://github.com/jonescompneurolab/hnn-core/pull/399/commits/263231802b9a74a5e6a3c39275bce620ddd2a6c3? I ended up converting the whole test to do what you suggested @ntolley as it makes the test more explicit and renders my original implementation redundant.

Reply:
This is great! Covers both questions with less code, that's a 10/10 in my book 😄